### PR TITLE
Add misc tests for System.Runtime.InteropServices

### PR DESCRIPTION
**Pull request by hughbe**
Final one in the Runtime.InteropServices backlog that can now be sent in without merge conflicts

### CODE REVIEW

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        [Fact]
+        public void Free_NotInitialized_ThrowsInvalidOperationException()
+        {
+            var handle = new GCHandle();
+            Assert.Throws<InvalidOperationException>(() => handle.Free());
+        }
+
+        public static IEnumerable<object[]> Equals_TestData()
+        {
+            var handle = GCHandle.Alloc(new object());
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs]: Who frees this?

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/SecureStringMarshalTests.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class SecureStringMarshalTests
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/SecureStringMarshalTests.cs]: How do these tests differ from or improve upon those here?
https://github.com/dotnet/corefx/blob/master/src/System.Security.SecureString/tests/SecureStringTests.cs

Comment (hughbe) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/SecureStringMarshalTests.cs]: They don't, i didn't see them in the project. Will remove

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs]: Updated to free this

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        public void AddrOfPinnedObject_NotInitialized_ThrowsInvalidOperationException()
+        {
+            var handle = new GCHandle();
+            Assert.Throws<InvalidOperationException>(() => handle.AddrOfPinnedObject());
+        }
+
+        [Fact]
+        public void AddrOfPinnedObject_NotPinned_ThrowsInvalidOperationException()
+        {
+            GCHandle handle = GCHandle.Alloc(new object());
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs]: This needs to be freed.

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                if (other is GCHandle otherHandle)
+                {
+                    Assert.Equal(expected, handle == otherHandle);
+                    Assert.Equal(!expected, handle != otherHandle);
+                }
+            }
+            finally
+            {
+                handle.Free();
+                if (other is GCHandle otherHandle && !ReferenceEquals(handle, otherHandle))
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs]: GCHandle is a struct.  `ReferenceEquals(handle, otherHandle)` will always return false, as the two structs will be boxed into different objects.

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs (hunk 3 above)
Comment (hughbe) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs]: ach, updated

**Comment by hughbe** [comment]
Windows 8.1 build failing with. Likely related
```
2017-06-24 02:53:21,039: INFO: proc(54): run_and_log_output: Output: Discovering: System.Runtime.InteropServices.Tests
2017-06-24 02:53:21,197: INFO: proc(54): run_and_log_output: Output: Discovered:  System.Runtime.InteropServices.Tests
2017-06-24 02:53:21,275: INFO: proc(54): run_and_log_output: Output: Starting:    System.Runtime.InteropServices.Tests
2017-06-24 02:53:23,668: INFO: proc(54): run_and_log_output: Output: Finished running tests.  End time= 2:53:23.66, Exit code = -1073741819
2017-06-24 02:53:23,668: INFO: proc(58): run_and_log_output: Exit Code: -1073741819
2017-06-24 02:53:23,668: ERROR: scriptrunner(82): _main: Error: No exception thrown, but XUnit results not created
2017-06-24 02:53:23,668: ERROR: helix_test_execution(83): report_error: Error running xunit None
```

**Comment by hughbe** [comment]
Test Windows x64 Debug Build

**Comment by danmosemsft** [comment]
Thanks @hughbe . LGTM

**Pull request merged**